Firmware/ultralcd.cpp

```diff
@@ -4525,8 +4525,7 @@ static void lcd_settings_menu()
     MENU_ITEM_TOGGLE_P(_T(MSG_RPI_PORT), (selectedSerialPort == 0) ? _T(MSG_OFF) : _T(MSG_ON), lcd_second_serial_set);
 #endif //HAS_SECOND_SERIAL
 
-	if ( babystep_allowed() )
-		MENU_ITEM_SUBMENU_P(_T(MSG_BABYSTEP_Z), lcd_babystep_z);
+    MENU_ITEM_SUBMENU_P(_T(MSG_BABYSTEP_Z), lcd_babystep_z);
```

Comment:
In 3.12 this was not allowed during homing or mesh bed leveling. Shouldnt that still apply?

Reply:
From looking at the code it seems to me the settings menu is accessible during homing unless `printJobOngoing()` returns `false`.

But I haven't checked this it on my printer.

Reply:
The Z adjust menu is dismissed immediately when opening it. So the current solution does not fix the menu.

I think editing is unavoidable `babystep_allowed()`
![image](https://user-images.githubusercontent.com/8218499/236295934-dccc8bfe-beb1-4a0a-ba3c-84103c441ea8.png)

Reply:
On the bright side this does cover mesh bed leveling and homing if that ever happens :)

Reply:
Here is my attempt at patching this. Seems to work.
[PR4188.patch](https://github.com/prusa3d/Prusa-Firmware/files/11401100/PR4188.patch)

Reply:
I like the patch but the result is that the Live adjust Z is also shown as 1st menu point
![Live_adjust_Z_in_main_menu](https://user-images.githubusercontent.com/25530011/236394656-9909ac43-6383-48dd-bb51-94f76155886a.png)

Reply:
@gudnimg @3d-gussner feel free to push directly to this branch if/when you find a solution

Reply:
I have a potential fix. Will push it once I have verified it tomorrow.